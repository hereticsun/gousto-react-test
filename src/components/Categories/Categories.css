.categories {

}

.categories__heading {
  
}

.categories__list {
  display: flex;
  flex-flow: row wrap;
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.categories__category {
  display: flex;
  flex: 0 0 auto;
  align-items: center;
  justify-content: center;
  margin: 8px;
  padding: 8px 16px;
  border-radius: 3px;
  color: #fff;
  background-color: #7290ff;
}
