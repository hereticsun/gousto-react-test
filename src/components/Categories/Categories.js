import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCategories } from '../../actions/categories';
import './Categories.css';

export class Categories extends Component {
  componentDidMount() {
    this.props.fetchCategories();
  }

  renderList() {
    return this.props.categories.map((category) => {
      return (
            <li
                key={category.id}
                className="categories__category"
            >
                {category.title}
            </li>
        )
    })
  }

  render() {
    return (
      <section className="categories">
        <h2 className="categories__heading">Categories:</h2>
        {this.props.categories.length > 0 ? (
          <ul className="categories__list">
              {this.renderList()}
          </ul>
        ) : null}
      </section>  
      )
  }
}

function mapStateToProps(state) {
  return {
      categories: state.categories
  };
}

Categories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      box_limit: PropTypes.number,
      is_default: PropTypes.bool,
      recently_added: PropTypes.bool,
      hidden: PropTypes.bool,
      pivot: PropTypes.shape({
        created_at: PropTypes.string,
      }),
    })),
};

export default connect(mapStateToProps, { fetchCategories })(Categories);
