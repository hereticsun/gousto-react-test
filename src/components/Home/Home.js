import React from 'react'
import { push } from 'connected-react-router'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import Categories from '../Categories';

export const Home = (props) => (
  <div>
    <Categories />
    <p>List products here</p>
  </div>
)

const mapStateToProps = state => state;

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      changePage: () => push('/about-us')
    },
    dispatch
  )

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home)
