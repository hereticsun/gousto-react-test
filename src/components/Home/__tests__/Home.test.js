import React from 'react';
import { shallow } from 'enzyme';

import { Home } from '../Home';
import Categories from '../../Categories';

describe('Home', () => {
  const element = shallow(<Home/>);

  it('renders properly', () => {
    expect(element).toMatchSnapshot();
  });

  it('should include the `Categories` component', () => {
    expect(element.find(Categories)).toHaveLength(1);
  });
});