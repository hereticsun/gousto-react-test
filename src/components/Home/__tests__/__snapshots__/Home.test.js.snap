// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Categories) />,
        <p>
          List products here
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "List products here",
        },
        "ref": null,
        "rendered": "List products here",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Categories) />,
          <p>
            List products here
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "List products here",
          },
          "ref": null,
          "rendered": "List products here",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
